src/sentry/snuba/metrics/fields/base.py

```diff
@@ -359,10 +359,23 @@ def run_post_query_function(
         else:
             subdata = data[alias][idx]
 
-        compute_func_dict = {"data": subdata}
-        if self.metrics_query_args is not None and params is not None:
-            for field in self.metrics_query_args:
-                compute_func_dict[field] = params.get(field)
+        # Fetch the function args
+        metrics_query_args = set(inspect.signature(self.post_query_func).parameters.keys())
```

Comment:
```suggestion
        metrics_query_args = inspect.signature(self.post_query_func).parameters.keys()
```
I think set is redundant here:
```
In [3]: def foo(a, b): pass

In [4]: inspect.signature(foo)
Out[4]: <Signature (a, b)>

In [5]: inspect.signature(foo).parameters
Out[5]: mappingproxy({'a': <Parameter "a">, 'b': <Parameter "b">})

In [6]: inspect.signature(foo).parameters.keys()
Out[6]: odict_keys(['a', 'b'])

In [7]: 1 in inspect.signature(foo).parameters.keys()
Out[7]: False
```

Reply:
fixed